### ISSUE TITLE
Problem populating the table with remote data.

### ISSUE THREAD

**Issue by pavelav** [arnedesmedt/vue-ads-table-tree#31]
Hello,
Thank you for the package!
I'm trying to use it to display server-side data, that I load over REST. In other words, initially the table is empty, but when the background ajax call completes, I'd like to display this data in the table.

*Approach 1*. I've tried using `call-rows` property, but to no avail. Looks like I need to know the number of rows in advance, which I don't with remote data. Do I get it right that this approach is just not suitable for the purpose, or am I doing something wrong?

*Approach 2* (updating `rows` property). I assume this is just not implemented yet (https://gitlab.com/arnedesmedt/vue-ads-table-tree/issues/33), though it would be much along the lines of how Vue works.
I've tried to take it just a bit further myself. Found that somehow `start` and `end` are set to `0` even though initialized to `null` in `VueAdsTableContainer`, and therefore in [pagination mixin](https://github.com/arnedesmedt/vue-ads-table-tree/blob/d5047b840daa7a8a73030740833be5ae758872bc/src/mixins/pagination.js#L20):
```javascript
        paginatedRows () {
            if (this.unresolved || (this.start === null && this.end === null)) {
                return this.sortedRows;
            }

            return this.sortedRows.slice(this.start, this.end);
        },
```
the condition is not met and we get a slice from 0 to 0 => empty array. If I change the condition to check for `0` in addition to `null`, the data is displayed in the table, but collapsing/expanding groups doesn't work. Must be something else to it.
As a workaround I've tried using a wrapper component to get the data, and then show the table component once data is available (via `v-if` with the data passed in as a param), but that didn't help either.

*Bottom line*
So is there any way I can get it to work with the current state of things, or do we need [the issue with `rows`](https://gitlab.com/arnedesmedt/vue-ads-table-tree/issues/33) solved first? Thank you!

Sincerely, Pavel

**Comment by arnedesmedt** [comment]
Hi Pavel,

_Approach 1_: Yes, you need to know the total number of rows. I've implemented this approach for an async pagination component. Maybe it's possible to query the total number of records on the server side when you load the component or just save the number in a var if it's server side rendered? 

The problem here is that you can't show a pagination component if you don't know the total number of rows. And in most cases it's also suitable to load small portions of the data and not all the records at once.

_Approach 2_: This is instead more the way Vue works (reactivity). But this feature is already implemented (don't check the gitlab repo, I know only work with the github version, otherwise it wasn't maintainable). It could be that there is a bug. I'll will check this later this week and check your PR you've made. Thx in advance to mention this.
